### PR TITLE
docs: remove direct references to PEP 517 in docs landing page

### PR DESCRIPTION
**Pull request by pradyunsg**
Toward #529

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -4,9 +4,10 @@
 build
 *****
 
-A simple, correct :pep:`517` build frontend.
+A simple, correct Python build frontend.
 
-build will invoke the :pep:`517` hooks to build a distribution package.
+build implements handling for pyproject.toml-based builds, invoking
```
Comment (layday) [inline docs/index.rst:9]: I think this is a bit convoluted, maybe:

```suggestion
build manages pyproject.toml-based builds, invoking
```

File: docs/index.rst [hunk 2]
```diff
@@ -4,9 +4,10 @@
...
+build-backend hooks as appropriate to build a distribution package.
```
Comment (layday) [inline docs/index.rst:10]: Is it possible to link to definitions of hooks and dist packages on pypa.io?

File: docs/index.rst (hunk 1 above)
Comment (pradyunsg) [reply docs/index.rst:9]: TBH, I mostly filed this as it seemed like an easy change wherein ~anything is better than status quo (which is plain wrong).

If we're word-smithing, I'd prefer that someone else drive this one over the line.

Comment (layday) [reply docs/index.rst:9]: We can merge this as-is and somebody else can copy-edit it in a subsequent PR if you prefer.

Comment (pradyunsg) [reply docs/index.rst:9]: That would work for me, or I'm happy for maintainers to push copy edits here directly. :)

File: docs/index.rst [hunk 3]
```diff
@@ -4,9 +4,10 @@
 build
 *****
 
-A simple, correct :pep:`517` build frontend.
+A simple, correct Python build frontend.
```
Comment (FFY00) [inline docs/index.rst:7]:
```suggestion
A simple, correct Python packaging build frontend.
```

This still feels a bit incomplete to me, but it's better.

File: docs/index.rst (hunk 1 above)
Comment (FFY00) [reply docs/index.rst:9]: We can do that, thank you for the initial contribution!

File: docs/index.rst (hunk 3 above)
Comment (pradyunsg) [reply docs/index.rst:7]: This PR matches what the README has. :)

File: docs/index.rst (hunk 2 above)
Comment (henryiii) [reply docs/index.rst:10]: Okay to do that in a followup?

Comment (layday) [reply docs/index.rst:10]: Yep.

File: docs/index.rst [hunk 4]
```diff
@@ -4,9 +4,10 @@
 build
 *****
 
-A simple, correct :pep:`517` build frontend.
+A simple, correct Python packaging build frontend.
 
-build will invoke the :pep:`517` hooks to build a distribution package.
+build manages pyproject.toml-based builds, invoking
```
Comment (FFY00) [inline docs/index.rst:9]:
```suggestion
build manages ``pyproject.toml``-based builds, invoking
```

**Review by FFY00 — APPROVED** [review]

File: docs/index.rst (hunk 2 above)
Comment (FFY00) [reply docs/index.rst:10]: Lets merge this and do it in a followup.

**Pull request merged**